File: .github/change_xcconfig.pl
use YAML;

open my $yaml_file, '<', './pubspec.yaml' or die "Cannot open file: $!";
my $data = YAML::LoadFile($yaml_file);
close $yaml_file;

my $version = substr($data->{version}, 0, -2);

open my $xcconfig_file, '<', "./ios/Flutter/Generated.xcconfig.orig" or die "Cannot open file: $!";
my @envs = <$xcconfig_file>;
close $xcconfig_file;

foreach my $env (@envs) {
    my ($key, $value) = split(/=/, $env, 2);
    if ($key eq "FLUTTER_BUILD_NAME") {
        $value = $version;
        $env = "$key=$value\n";
    }
}

open $xcconfig_file, '>', "./ios/Flutter/Generated.xcconfig" or die "Cannot open file: $!";
print $xcconfig_file @envs;
close $xcconfig_file;

File: .github/change_xcconfig.rb
require 'yaml'

data = open('./pubspec.yaml', 'r') { |f| YAML.load(f) }
version = data["version"][0..-3]

envs = open("./ios/Flutter/Generated.xcconfig").read()
envs = envs.split("\n").map{|e| e.split("=") }
envs.map! do |e|
  if e[0] == "FLUTTER_BUILD_NAME"
    e[1] = version
  end
  e
end
envs = envs.map{|e| e.join("=") }.join("\n")
open("./ios/Flutter/Generated.xcconfig", "w").write(envs)
